File: maximum-subarray.java
// If the sum is negative, reset to 0. And keep updating the maxSum at each iteration.

class Solution {
  public int maxSubArray(int[] nums) {
  int curSum=0;
  int maxSum= nums[0];
  for(int i=0;i<nums.length;i++){
    if (curSum<0){
      curSum=0;
    }
    curSum=curSum+nums[i];
    if (curSum>maxSum){
      maxSum=curSum;
    }
  }
  return maxSum;
  }
}

File: maximum_subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cursum=0
        maxsum=nums[0]

        for num in nums:
            cursum=cursum+num

            if(cursum>maxsum):
                maxsum=cursum

            if cursum<0:
                cursum=0

        return maxsum
